pkg/cmd/projectconfig/add.go

```diff
@@ -36,91 +35,114 @@ var projectConfigAddCmd = &cobra.Command{
-		create.ProjectsConfigurationChanged, err = create.RunProjectConfiguration(&createDtos, *projectDefaults)
-		if err != nil {
-			log.Fatal(err)
-		}
-
-		if len(createDtos) == 0 {
-			log.Fatal("no projects found")
-		}
-
-		if createDtos[0].Name == "" {
-			log.Fatal("project config name is required")
-		}
-
-		initialSuggestion := createDtos[0].Name
-
-		chosenName := workspace_util.GetSuggestedName(initialSuggestion, existingProjectConfigNames)
-
-		submissionFormConfig := create.SubmissionFormConfig{
-			ChosenName:    &chosenName,
-			SuggestedName: chosenName,
-			ExistingNames: existingProjectConfigNames,
-			ProjectList:   &createDtos,
-			NameLabel:     "Project config",
-			Defaults:      projectDefaults,
-		}
-
-		err = create.RunSubmissionForm(submissionFormConfig)
+		projectConfig, err := RunProjectConfigAddFlow(apiClient, gitProviders, ctx)
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		newProjectConfig := apiclient.CreateProjectConfigDTO{
-			Name:          chosenName,
-			BuildConfig:   createDtos[0].BuildConfig,
-			Image:         createDtos[0].Image,
-			User:          createDtos[0].User,
-			RepositoryUrl: createDtos[0].Source.Repository.Url,
-		}
-
-		newProjectConfig.EnvVars = *workspace_util.GetEnvVariables(createDtos[0].EnvVars, nil)
-
-		res, err = apiClient.ProjectConfigAPI.SetProjectConfig(ctx).ProjectConfig(newProjectConfig).Execute()
-		if err != nil {
-			log.Fatal(apiclient_util.HandleErrorResponse(res, err))
+		if projectConfig != nil {
+			views.RenderInfoMessage("Project config added successfully")
 		}
+	},
+}
 
-		views.RenderInfoMessage("Project config added successfully")
+func RunProjectConfigAddFlow(apiClient *apiclient.APIClient, gitProviders []apiclient.GitProvider, ctx context.Context) (*apiclient.ProjectConfig, error) {
+	var createDtos []apiclient.CreateProjectDTO
+	var existingProjectConfigNames []string
+
+	apiServerConfig, res, err := apiClient.ServerAPI.GetConfig(context.Background()).Execute()
+	if err != nil {
+		return nil, apiclient_util.HandleErrorResponse(res, err)
+	}
+
+	existingProjectConfigs, res, err := apiClient.ProjectConfigAPI.ListProjectConfigs(context.Background()).Execute()
+	if err != nil {
+		return nil, apiclient_util.HandleErrorResponse(res, err)
+	}
+	for _, pc := range existingProjectConfigs {
+		existingProjectConfigNames = append(existingProjectConfigNames, pc.Name)
+	}
+
+	projectDefaults := &create.ProjectConfigDefaults{
+		BuildChoice:          create.AUTOMATIC,
+		Image:                &apiServerConfig.DefaultProjectImage,
+		ImageUser:            &apiServerConfig.DefaultProjectUser,
+		DevcontainerFilePath: create.DEVCONTAINER_FILEPATH,
+	}
+
+	createDtos, err = workspace_util.GetProjectsCreationDataFromPrompt(workspace_util.ProjectsDataPromptConfig{
+		UserGitProviders:    gitProviders,
+		Manual:              manualFlag,
+		MultiProject:        false,
+		SkipBranchSelection: true,
+		ApiClient:           apiClient,
+		Defaults:            projectDefaults,
 	},
+	)
```

Comment:
Just a note to maybe move this `)` up a line and remove the coma after `}` on line 80